packages/e2e-test/src/e2e-test.ts

```diff
@@ -273,28 +273,36 @@ async function testAppServe(pluginName: string, appDir: string) {
   const startApp = spawnPiped(['yarn', 'start'], {
     cwd: appDir,
   });
-  Browser.localhost('localhost', 3000);
 
   let successful = false;
-  try {
-    const browser = new Browser();
+  for (let attempts = 1; ; attempts++) {
+    Browser.localhost('localhost', 3000);
 
-    await waitForPageWithText(browser, '/', 'Backstage Service Catalog');
-    await waitForPageWithText(
-      browser,
-      `/${pluginName}`,
-      `Welcome to ${pluginName}!`,
-    );
+    try {
+      const browser = new Browser();
```

Comment:
does this pick a dynamic port and whatnot?

Reply:
Nah, this just browses localhost:3000